src/checkpoint/checkpoint_txn.c

```diff
@@ -856,9 +856,10 @@ __checkpoint_prepare(WT_SESSION_IMPL *session, bool *trackingp, const char *cfg[
     /*
      * Save the checkpoint session ID.
      *
-     * We never do checkpoints in the default session (with id zero).
+     * We never do checkpoints in the default session.
      */
-    WT_ASSERT(session, session->id != 0 && __wt_atomic_loadv32(&txn_global->checkpoint_id) == 0);
+    WT_ASSERT(session,
+      !WT_IS_DEFAULT_SESSION(session) && __wt_atomic_loadv32(&txn_global->checkpoint_id) == 0);
```

Comment:
Sorry @lukech, another nit: as part convention we should have WT_{module_name}_xxx. Should we change this to `WT_SESSION_IS_DEFAULT` (I should have mentioned it before)?